features/filesystem/bd/ChainingBlockDevice.cpp

```diff
@@ -211,6 +211,22 @@ bd_size_t ChainingBlockDevice::get_erase_size() const
     return _erase_size;
 }
 
+bd_size_t ChainingBlockDevice::get_erase_size(bd_addr_t addr) const
+{
+    bd_addr_t bd_start_addr = 0;
+    for (size_t i = 0; i < _bd_count; i++) {
+        bd_size_t bdsize = _bds[i]->size();
+        if (addr < (bd_start_addr + bdsize)) {
+            return _bds[i]->get_erase_size(addr - bd_start_addr);
+        }
+        bd_start_addr += bdsize;
+    }
+
+    // Getting here implies an illegal address
+    MBED_ASSERT(0);
+    return 0; // satisfy compiler
+}
+
```

Comment:
Yeah, this looks good :+1:

Reply:
Nothing to block this PR: Just queries for my understanding 
1. I see `_bd_count(sizeof(bds) / sizeof(bds[0])` set during constructor, does this mean we erase all chain blocks in erase operation?

Reply:
An erase can cross multiple block devices in the chain.

Here's where it's implemented:
https://github.com/davidsaada/mbed-os/blob/a6048005d7c7ee06b7956918589d381a9f41f159/features/filesystem/bd/ChainingBlockDevice.cpp#L174-L194